packages/devtools/lib/src/timeline/cpu_profile_model.dart

```diff
@@ -171,8 +178,8 @@ class CpuStackFrame extends TreeNode {
 
   Duration _totalTime;
 
-  double get selfTimeRatio => _selfTimeRatio ??=
-      exclusiveSampleCount / (root as CpuStackFrame).inclusiveSampleCount;
+  double get selfTimeRatio =>
+      _selfTimeRatio ??= exclusiveSampleCount / profileSampleCount;
```

Comment:
why is this getter and the _totalTimeRatio cached? I expect the caching might actually have a negative perf impact and could lead to bugs if the sampleCount or exclusiveSampleCount changed.